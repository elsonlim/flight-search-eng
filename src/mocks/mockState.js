import moment from 'moment';

export const flights = [
    {
        "id": 0,
        "IATA": "DL",
        "flightNo": 90,
        "from": "USA",
        "to": "HKG",
        "price": {"amount": 261, "currency": "USD"},
        "time": {"arrival": 609, "departure": 425, "duration": 104}
    }, {
        "id": 1,
        "IATA": "DL",
        "flightNo": 91,
        "from": "HKG",
        "to": "USA",
        "price": {"amount": 232, "currency": "USD"},
        "time": {"arrival": 1956, "departure": 1730, "duration": 146}
    },
];

export const searchParams = {
    isReturn: true,
    from: 'USA',
    to: 'HKG',
    depatureDate: moment(new Date()),
    arrivalDate: moment(new Date()),
    numberOfPassengers: 1,
}

export const priceRange = {
    filter: {min: 0, max: 2000},
    search: {min: 200, max: 1000},
}

export default {
    flights,
    searchType: false,
    searchParams,
    priceRange
};

    