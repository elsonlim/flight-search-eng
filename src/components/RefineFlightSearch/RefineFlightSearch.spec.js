import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import enzyme, { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import Adapter from 'enzyme-adapter-react-16';
import configureStore from 'redux-mock-store';
import mockState from '../../mocks/mockState';
import { RefineFlightSearch } from './index';

enzyme.configure({ adapter: new Adapter() });

describe('RefineFlightSearch', () => {

    let props;
    let wrapper;
    let store;
    const mockStore = configureStore();

    beforeEach(() => {
        props = Object.assign({}, mockState);
        props.updatePriceSearch = jest.fn();
        store =  mockStore(mockState);
        wrapper = shallow(<RefineFlightSearch {...props} />);
    });
    
    describe('snapshot test', () => {
		test('show refine flight search', () => {
			const tree = renderer.create(<Provider store={store}><RefineFlightSearch {...props} /></Provider>).toJSON();
			expect(tree).toMatchSnapshot();
		});
    });

    describe('updatePriceFilter', () => {
        it('calls props.updatePriceRange with value', () => {
            wrapper.instance().updatePriceFilter('price', 123);
            //console.log(props);
            expect(props.updatePriceSearch).toHaveBeenCalledTimes(1);
        });
    });
});
