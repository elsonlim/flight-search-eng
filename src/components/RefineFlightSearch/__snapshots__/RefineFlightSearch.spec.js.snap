// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RefineFlightSearch snapshot test show refine flight search 1`] = `
<div
  className="ui fluid card"
  href={undefined}
  onClick={[Function]}
>
  <div
    className="content"
  >
    <div
      className="left aligned header"
    >
      Refine Flight Search
    </div>
    <div
      className="refine-search-slider"
    >
      <div
        style={
          Object {
            "marginBottom": "10px",
            "textAlign": "left",
          }
        }
      >
        <div
          style={
            Object {
              "display": "flex",
              "flexWrap": "wrap",
              "justifyContent": "space-between",
            }
          }
        >
          <span
            style={
              Object {
                "fontWeight": "bold",
                "marginRight": "10px",
              }
            }
          >
            Price/Pax
          </span>
          <span
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#1678C2",
                "border": "1px solid #e9e9e9",
                "borderRadius": "5px",
                "color": "white",
                "display": "flex",
                "fontSize": "0.8em",
                "fontWeight": "normal",
                "padding": "0 2px",
                "whiteSpace": "nowrap",
              }
            }
          >
            200
             - 
            1000
          </span>
        </div>
        <div
          style={
            Object {
              "padding": "5px",
            }
          }
        >
          <div
            className="rc-slider"
            onBlur={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            onTouchStart={[Function]}
            style={undefined}
          >
            <div
              className="rc-slider-rail"
              style={
                Object {
                  "backgroundColor": "grey",
                }
              }
            />
            <div
              className="rc-slider-track rc-slider-track-1"
              style={
                Object {
                  "backgroundColor": "#1678C2",
                  "left": "10%",
                  "width": "40%",
                }
              }
            />
            <div
              className="rc-slider-step"
            />
            <div
              aria-disabled={false}
              aria-valuemax={2000}
              aria-valuemin={0}
              aria-valuenow={200}
              className="rc-slider-handle rc-slider-handle-1"
              onBlur={undefined}
              onClick={undefined}
              onContextMenu={undefined}
              onFocus={undefined}
              onMouseDown={undefined}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onTouchStart={undefined}
              role="slider"
              style={
                Object {
                  "borderColor": "#1678C2",
                  "left": "10%",
                }
              }
              tabIndex={0}
            />
            <div
              aria-disabled={false}
              aria-valuemax={2000}
              aria-valuemin={0}
              aria-valuenow={1000}
              className="rc-slider-handle rc-slider-handle-2"
              onBlur={undefined}
              onClick={undefined}
              onContextMenu={undefined}
              onFocus={undefined}
              onMouseDown={undefined}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onTouchStart={undefined}
              role="slider"
              style={
                Object {
                  "borderColor": "#1678C2",
                  "left": "50%",
                }
              }
              tabIndex={0}
            />
            <div
              className="rc-slider-mark"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
